nymea-app/ui/mainviews/WallboxView.qml

```diff
@@ -46,70 +46,94 @@ MainViewBase {
         shownInterfaces: ["evcharger"]
     }    
 
-    SwipeView {
-        id: swipeView
+    Rectangle {
         anchors.fill: parent
-        anchors.horizontalCenter: parent.horizontalCenter
-        currentIndex: pageIndicator.currentIndex
-        visible: wallboxDevices.count != 0
+        color: app.foregroundColor
 
-        Repeater {
+        SwipeView {
+            id: swipeView
             anchors.fill: parent
             anchors.horizontalCenter: parent.horizontalCenter
-            model: wallboxDevices
-
-            delegate: Item {
-                id: wallboxDelegate
-                height: swipeView.height
-                width: swipeView.width
-
-                property Thing thing: wallboxDevices.get(index)
-                property State powerState: thing.stateByName("power")
-                property State maxChargingCurrentState: thing.stateByName("maxChargingCurrent")
-
-                Rectangle {
-                    width: swipeView.width
+            currentIndex: pageIndicator.currentIndex
+            visible: wallboxDevices.count != 0
+
+            Repeater {
+                anchors.fill: parent
+                anchors.horizontalCenter: parent.horizontalCenter
+                model: wallboxDevices
+                delegate: Item {
+                    id: wallboxDelegate
                     height: swipeView.height
-                    anchors.horizontalCenter: parent.horizontalCenter
+                    width: swipeView.width
 
-                    GridLayout {
-                        anchors.fill: parent
-                        anchors.margins: app.margins
+                    property Thing thing: wallboxDevices.get(index)
+                    property State powerState: thing.stateByName("power")
+                    property State maxChargingCurrentState: thing.stateByName("maxChargingCurrent")
+
+                    ColumnLayout {
+                        width: swipeView.width
+                        height: swipeView.height
                         anchors.horizontalCenter: parent.horizontalCenter
-                        columns: 1
-                        rowSpacing: app.margins
 
-                        Text {
-                            anchors.horizontalCenter: parent.horizontalCenter
-                            text: thing.name
+                        Rectangle {
+                            Layout.alignment: Qt.AlignCenter
+                            Layout.preferredWidth: thingName.width * 1.5
+                            Layout.preferredHeight: thingName.height * 2
```

Comment:
You probably want to use implicitWidth/implicitHeight when asking a child item for preferred... The difference being:
* width/height is the actual size (which might be smaller/larger than the text content)
* implicitWidth/implicitHeight is the size that an item would have if noone tells it to resize, in other words, what the label would like to be ideally